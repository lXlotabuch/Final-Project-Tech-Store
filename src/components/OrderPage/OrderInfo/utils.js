/* eslint-disable import/prefer-default-export */
export const getDate = (date) => date.slice(0, 10).split('-').reverse().join('.')