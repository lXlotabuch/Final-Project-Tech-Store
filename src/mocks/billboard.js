/* eslint-disable import/prefer-default-export */
export const billboardMock = {
  title: 'gaming monitors',
  img: 'https://fuzzmusic.ru/wp-content/uploads/2020/08/pesni-iz-igry-kiberpank-2077-1.jpg',
  url: '/catalog?categories=gamingMonitors',
  category: 'gaming monitors'
}