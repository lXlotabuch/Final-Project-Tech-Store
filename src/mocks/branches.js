/* eslint-disable import/prefer-default-export */
export const branchesMock = [
  {
    branchName: 'Branch name',
    branchRef: 'kjhkjh3j23kjhkjhk32jhkj4',
  }
]