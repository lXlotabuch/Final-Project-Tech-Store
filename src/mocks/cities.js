/* eslint-disable import/prefer-default-export */
export const citiesMock = [
  {
    CityName: 'Kyiv',
    Ref: '8d5a980d-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Dnipro',
    Ref: 'db5c88f0-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Kharkiv',
    Ref: 'db5c88e0-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Lviv',
    Ref: 'db5c88f5-391c-11dd-90d9-001a92567626'
  },
  {
    CityName: 'Odessa',
    Ref: 'db5c88d0-391c-11dd-90d9-001a92567626'
  },
]

export const cityMock = {
  CityName: 'Kyiv',
  Ref: '8d5a980d-391c-11dd-90d9-001a92567626'
}