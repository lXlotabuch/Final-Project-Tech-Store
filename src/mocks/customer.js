export const mockCustomer = {
  customerNo: '82724063',
  date: '2021-02-19T17:12:50.176+00:00',
  _id: '602ff192ad29b70015add9ab',
  email: 'o.titova.ua@gmail.com',
  isAdmin: false,
  enabled: true,
  firstName: 'someName',
  lastName: 'someLastName',
  login: 'admin',
  password: '$2a$10$i7f6tUO3IT.YsrEGzo/6neTIzoxjhhpWh3Hw3Cme3KLmIAeeyKtFS',
  telephone: '+380731005009',
  gender: 'male',
  ratedProducts: [
    '33823',
    '296354',
    '996598',
    '500227',
    '694186',
    '73109',
    '394414',
    '833228',
    '562546',
    '954601',
    '179158',
    '132036'
  ],
}
export default mockCustomer
