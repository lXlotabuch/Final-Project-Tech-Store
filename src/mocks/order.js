/* eslint-disable import/prefer-default-export */
export const orderMock = {
  deliveryAddress: {country: 'Ukraine', city: 'db5c88f0-391c-11dd-90d9-001a92567626', branch: 'Отделение №2: ул. Академика Янгеля, 40'},
  email: 'o.titova.ua@gmail.com',
  firstName: 'test',
  letterHtml: '<h1>Your order is placed. Our manager will contact you soon.</h1>',
  letterSubject: 'test, thank you for order!',
  mobile: '+380660788828',
  orderNo: '8395086',
  paymentInfo: 'Payment at the time of receipt of the goods',
  shipping: 121,
  totalSum: 117400,
  date: '2021-04-13T14:14:27.478Z',
  _id: '6070d130058f4e00154bbe23'
}