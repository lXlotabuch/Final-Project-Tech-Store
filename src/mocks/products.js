export const productMock = {
  _id: '604ffa554e7ef500153393f9',
  name: 'Monitor 27" AOC C27G2ZU/BK',
  enabled: true,
  currentPrice: 10600,
  previousPrice: 12600,
  categories: 'gamingMonitors',
  imageUrls: [
    'https://i8.rozetka.ua/goods/18842039/aoc_c27g2zu_bk_images_18842039923.jpg',
    'https://i2.rozetka.ua/goods/18842040/aoc_c27g2zu_bk_images_18842040211.jpg',
    'https://i8.rozetka.ua/goods/18842040/aoc_c27g2zu_bk_images_18842040937.jpg',
    'https://i8.rozetka.ua/goods/18842041/aoc_c27g2zu_bk_images_18842041729.jpg'
  ],
  quantity: 44,
  color: 'black',
  productUrl: '/gamingMonitors',
  brand: 'AOC',
  newProduct: 'no',
  reviews: [4, 2, 3, 5, 4, 5, 3, 4, 3, 5, 5, 4, 5],
  params: {
    updateFrequency: '240Hz',
    countryOfmanufacture: 'China',
    displayDiagonal: '27"',
    displayResolution: '1920x1080',
    responseTime: '0,5ms',
    screenSize: '32 inches',
    aspectRatio: '16:9'
  },
  description: 'An immersive 27" super curved gaming monitor with a fast 144Hz refresh rate and 1ms response time, helping you get the most out of your monitor. Features such as AMD FreeSync Premium, QHD resolution, and a curved design make this monitor ultra competitive for any gamer.',
  itemNo: '554194',
}
const productMock2 = {
  ...productMock,
  itemNo: '554195',
  _id: '604ffa554e7ef500153393f0',
}
export const productsMock = [
  {
    _id: '607057d9e448c200152a5047',
    cartQuantity: 1,
    product: productMock
  },
  {
    _id: '607057d9e448c200152a5048',
    cartQuantity: 1,
    product: productMock2
  }
];