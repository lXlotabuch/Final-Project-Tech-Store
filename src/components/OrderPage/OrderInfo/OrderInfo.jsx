import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { Alert, Spin } from 'antd';
import { Redirect } from 'react-router-dom';
import StyledOrderInfo from './StyledOrderInfo';
import { selectCities, selectOrder, selectIsLoading } from '../../../store/cart/reducer';
import { getDate } from './utils';

export const OrderInfoComponent = ({order, cities, isLoading}) => {
  const date = order.date ? getDate(order.date) : null

  const cityName = (cityRefForShipping) => {
    const cityCustomer = cities.find((item) => item.Ref === cityRefForShipping)
    return cityCustomer.CityName
  }

  const showOrderInfo = (newOrder) => (
    <div>
      <h2>
        <span className="italic">
          {newOrder.firstName}
        </span>
        , your order has been accepted.
      </h2>
      <h2>Thank you for your purchase.</h2>
      <p>
        Your orderNo is:
        <span>
          {newOrder.orderNo}
        </span>
      </p>
      <p>
        Order total:
        <span>
          {newOrder.totalSum}
          â‚´
        </span>
      </p>
      <p>
        Delivery city:
        <span>
          {cityName(newOrder.deliveryAddress.city)}
        </span>
      </p>
      <p>
        Delivery branch:
        <span>
          {newOrder.deliveryAddress.branch}
        </span>
      </p>
      <p>
        Payment method:
        <span>
          {newOrder.paymentInfo}
        </span>
      </p>
      <p>
        Date:
        <span>
          {date}
        </span>
      </p>
      <h2>
        Our manager will contact you soon.
      </h2>
    </div>
  )
  const isOrder = Boolean(Object.keys(order).length === 0)
  if (isOrder && !isLoading) return <Redirect to="/" />
  return (
    <StyledOrderInfo>
      {isLoading
        ? (
          <Spin tip="Loading..." size="large">
            <Alert
              message="Please wait..."
              description="Order information will appear on screen soon"
              type="info"
            />
          </Spin>
        )
        : showOrderInfo(order) }
    </StyledOrderInfo>
  )
}

const mapStateToProps = (state) => ({
  order: selectOrder(state),
  cities: selectCities(state),
  isLoading: selectIsLoading(state),
})

const OrderInfo = connect(mapStateToProps, null)(OrderInfoComponent)

OrderInfoComponent.propTypes = {
  isLoading: PropTypes.bool.isRequired,
  order: PropTypes.shape({
    orderNo: PropTypes.string,
    totalSum: PropTypes.number,
    paymentInfo: PropTypes.string,
    date: PropTypes.oneOfType([PropTypes.number, PropTypes.string])
  }).isRequired,
  cities: PropTypes.arrayOf(
    PropTypes.shape({
      CityName: PropTypes.string.isRequired,
      Ref: PropTypes.string.isRequired,
    })
  ).isRequired,
}

export default OrderInfo