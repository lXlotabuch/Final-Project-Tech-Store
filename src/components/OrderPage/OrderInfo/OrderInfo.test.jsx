import React from 'react';
import { render } from '@testing-library/react';
import { citiesMock } from '../../../mocks/cities';
import { orderMock } from '../../../mocks/order';
import { OrderInfoComponent } from './OrderInfo';

describe('OrderInfoComponent', () => {
  test('render', () => {
    const {asFragment} = render(<OrderInfoComponent
      order={orderMock}
      cities={citiesMock}
      isLoading={false}
    />)
    expect(asFragment()).toMatchSnapshot()
  })
})