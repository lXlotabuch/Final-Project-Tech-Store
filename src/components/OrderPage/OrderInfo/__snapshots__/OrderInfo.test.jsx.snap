// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderInfoComponent render 1`] = `
<DocumentFragment>
  <div
    class="sc-bdfBwQ jPtdYN"
  >
    <div>
      <h2>
        <span
          class="italic"
        >
          test
        </span>
        , your order has been accepted.
      </h2>
      <h2>
        Thank you for your purchase.
      </h2>
      <p>
        Your orderNo is:
        <span>
          8395086
        </span>
      </p>
      <p>
        Order total:
        <span>
          117400₴
        </span>
      </p>
      <p>
        Delivery city:
        <span>
          Dnipro
        </span>
      </p>
      <p>
        Delivery branch:
        <span>
          Отделение №2: ул. Академика Янгеля, 40
        </span>
      </p>
      <p>
        Payment method:
        <span>
          Payment at the time of receipt of the goods
        </span>
      </p>
      <p>
        Date:
        <span>
          13.04.2021
        </span>
      </p>
      <h2>
        Our manager will contact you soon.
      </h2>
    </div>
  </div>
</DocumentFragment>
`;
